features/FxA-89-devices-view/README.md

```diff
@@ -0,0 +1,413 @@
+
+The link labelled "Missing or duplicate devices?"
+will link to a SUMO article:
+
+* https://support.mozilla.org/kb/fxa-managing-devices
+
+When a user is using desktop Firefox
+and has a web session with accounts.firefox.com
+but is not connected with the current browser,
+we can provide a "Connect..." button that links to:
+
+* /signin?service=sync&context=fx_desktop_v3"
+
+
+## Implementation Plan
+
+We will implement this feature in several phases,
+intending each phase to be independently shippable
+if we so choose.
+
+Since the primary goal for this feature is security,
+we will first focus on implementing the parts
+related to disconnecting devices.
+This will be followed by additional notifications
+and security checks to improve the disconnection experience,
+and finally by the new affordances
+for connecting additional devices.
+
+### Phase 1
+
+First we'll make sure
+that the core disconnecting-your-devices experience
+works well and emits the necessary metrics,
+building on the existing devices view code.
+
+Acceptance criteria:
+
+* [ ] When I am connected to Sync on Desktop Firefox,
+      my device list shows that device
+      with a desktop icon,
+      a plausible last-active timestamp,
+      and a "Disconnect..." button.
+* [ ] When I am connected to Sync from Fennec or iOS on a phone,
+      my device list shows that device
+      with a phone icon,
+      a plausible last-active timestamp,
+      and a "Disconnect..." button.
+* [ ] When I am connected to Sync from Fennec or iOS on a tablet,
+      my device list shows that device
+      with a tablet icon,
+      a plausible last-active timestamp,
+      and a "Disconnect..." button.
+* [ ] When I sign in to Firefox Accounts on the web
+      without connecting to sync,
+      this does not appear as an item
+      in the device list.
+* [ ] The devices view does not show
+      any messaging about connecting new devices.
+* [ ] Clicking the "Disconnect..." button for a device
+      causes a modal confirmation dialog to be displayed
+      with the specified messaging.
+* [ ] Successfully submitting the dialog
+      destroys the device record and session
+      token for that device.
+* [ ] When I disconnect my current Desktop device
+      through the devices view,
+      and then go to about:preferences#sync,
+      I immediately see that the device is disconnected.
+* [ ] When I disconnect a different Desktop device
+      through the devices view,
+      and then go to about:preferences#sync on that device,
+      I immediately see that the device is disconnected.
+* [ ] Front-end metrics events are emitted
+      when users view the devices list,
+      click the "Disconnect..." button,
+      submit or cancel the disconnect dialog,
+      click the "refresh" button,
+      and click the "missing or duplicate devices" link.
+* [ ] Back-end metrics events are emitted
+      when a device is disconnected,
+      and include the reason for its disconnection.
```

Comment:
@rfk any reason why you would prefer Back-end metrics to include reason? 

We can include the `reason` in the front-end statsd metrics.